source/server/listener_manager_impl.cc

```diff
@@ -88,19 +90,47 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, ListenerManag
       local_drain_manager_(parent.factory_.createDrainManager()) {
   // TODO(htuch): Support multiple filter chains #1280, add constraint to ensure we have at least on
   // filter chain #1308.
-  ASSERT(config.filter_chains().size() == 1);
-  const auto& filter_chain = config.filter_chains()[0];
-
-  listener_scope_ =
-      parent_.server_.stats().createScope(fmt::format("listener.{}.", address_->asString()));
+  ASSERT(config.filter_chains().size() >= 1);
+
+  // Skip lookup and update of the SSL Context if there is only one filter chain
```

Comment:
Does this result in a noticable performance boost?  I didn't think that a few hash table lookups and string searching calls would be noticable compared to the TLS handshake.

Reply:
It's unlikely that there is any measurable performance boost, I added it to avoid locking.